### ISSUE TITLE
MySQL/MySQL2 Dependency Confusion

### ISSUE THREAD

**Issue by unisys12** [sequelize/sequelize#6546]
## What you are doing?
So, this is going to sound really stupid and I am a little weary to post this, but... after searching and digging, I felt I wanted to bring it up - if nothing more than as a learning experience. 

I decided to play around with Sequelize this morning and just going over the docs and [Getting Started](http://docs.sequelizejs.com/en/latest/docs/getting-started/) example, I ran into a slight issue. The _Getting Started_ example clearly states that after installing the core package (Sequelize) you should install a separate package so that you can work with whichever database you choose. In my case, MySQL. Well, the package it tells me to install is _mysql2_, so I do. But when I try to run the `auth test` outlined below, I receive an error. At this point, I will let the template take over...  

```js
var Sequelize = require('sequelize');

// Set-up the Sequalize Client
var db = new Sequelize(process.env.DB_DATABASE, process.env.DB_USERNAME, process.env.DB_PASSWORD, {
  host: process.env.DB_HOST,
  dialect: 'mysql',

  pool: {
    max: 5,
    min: 0,
    idle: 10000
  }

});

db.authenticate()
  .then(function(err) {
    console.log('Connection has been established successfully.');
  })
  .catch(function (err) {
    console.log('Unable to connect to the database:', err);
  });
```

## What do you expect to happen?
```'Connection has been established successfully.'```

## What is actually happening?
```bash
C:\sites\sequel_test\node_modules\sequelize\lib\dialects\mysql\connection-manager.js:24
      throw new Error('Please install mysql package manually');
      ^

Error: Please install mysql package manually
    at new ConnectionManager (C:\sites\sequel_test\node_modules\sequelize\lib\dialects\mysql
\connection-manager.js:24:13)
    at new MysqlDialect (C:\sites\sequel_test\node_modules\sequelize\lib\dialects\mysql\inde
x.js:12:28)
```

Upon inspecting `C:\sites\sequel_test\node_modules\sequelize\lib\dialects\mysql
\connection-manager.js:24:13` I can clearly see that the package defaults to `require('mysql')` and not `mysql2`. Changing this line in the above file, clears up the error of course, which of course we all know is a big __NO-NO!__. The best way to solve this is to install the `mysql` package through NPM and all is good in the world again. 

With that said, not sure if the Docs should be updated to reflect that the end-user should install `mysql` dependency and not `mysql2` or the package (connection-manager.js:20:27) should be updated to `mysql2`. 

__Dialect:__ mysql 

__Sequelize version:__ 3.24.1

**Comment by sushantdhiman** [comment]
I think you were checking master version of docs. If you look at http://docs.sequelizejs.com/en/v3/docs/getting-started/ , its still says `mysql` not `mysql2`.

But for latest `master` `v4` version you will need to use `mysql2` :)

**Comment by unisys12** [comment]
Totally makes sense... DOH! I guess we are all allowed at least one Homer Simpson moment a day right?

**Comment by sushantdhiman** [comment]
Right :)

**Comment by bluishoul** [comment]
@sushantdhiman setting a default document version(v3) will reduce this kind of issue?

**Comment by sushantdhiman** [comment]
@bluishoul I think v3 is default version for docs now.

**Comment by gihrig** [comment]
>  I think v3 is default version for docs now.

Not exactly...

**tldr**: The confusion comes from a mismatch between the npm package v3 and the `master` version docs in this repo 'v4'.

Although this is documented in `README.md` and the repo slug points to v3 docs, there just too many opportunities find and follow v4 docs with v3 code.

---

This documents my frustrating search though docs links. It can be ignored pretty much by anyone whole is not engaged in fixing the problem.

The [npm page](https://www.npmjs.com/package/sequelize) contains a clear link to "Documentation" `http://sequelize.readthedocs.org/en/latest/` which contains the `mysql2` requirement.

This repo's `README.md` (master branch) `getting Started` link references `http://docs.sequelizejs.com/en/latest/docs/getting-started/` which also has the `mysql2` requirement.

Even the `Getting Started` guide at `https://github.com/sequelize/sequelize/blob/master/docs/docs/getting-started.md` refers to the `mysql2` dependency.

The correct (v3) `Getting Started` guide can be found in the docs listed in the 'slug' at the top of this repo, `http://sequelize.readthedocs.io/en/v3/docs/getting-started/` This is the only link I have seen to this version.

**Comment by kahwooi** [comment]
Should I use mysql or mysql2?

**Comment by xpioneer** [comment]
"Error: Please install mysql2 package manually"
sequelize:4.8.3
I have encountered this problem。I installed "mysql2", It's working!

**Comment by MattMurn** [comment]
Had this issue recently. Thanks for posting

**Comment by Fikifh** [comment]
i have the same problems, but i have install mysq2 and then error yet. why that occur?

**Comment by jiradeto** [comment]
`npm install mysql2 -g`

**Comment by santosh9991** [comment]
npm install --save mysql2 works for me

**Comment by eingurat504** [comment]
npm install mysql2 worked for me
